packages/@ember/-internals/utils/lib/intern.ts

```diff
@@ -37,8 +37,8 @@
   @private
   @return {String} interned version of the provided string
 */
-export default function intern(str: string): string {
-  let obj: Record<string, number> = {};
+export default function intern<S extends string>(str: S): S {
+  let obj: Record<S, number> = Object.create(null);
```

Comment:
To keep this equivalent to `{}`:
```suggestion
  let obj: Record<S, number> = Object.create(Object.prototype);
```
`Object.create(null)` will have _no_ prototype, but `{}` has `Object`'s prototype:

<img width="249" alt="Screen Shot 2022-11-15 at 7 39 21 PM" src="https://user-images.githubusercontent.com/348630/202055013-9296b432-92a1-48c7-bf1d-83c794543353.png">

Reply:
Or just cast? ¯\\\_(ツ)\_/¯
```suggestion
  let obj = {} as Record<S, number>;
```

Reply:
Yeah, no prototype is actually fine here. I ran this it @rwjblue and the behavioral change here is totally fine; I just wanted it in a separate PR since it is a behavioral and not purely types change.